front/src/App/RecoveryEmail/RecoveryEmail.jsx

```diff
@@ -1,27 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Box, Notification } from 'grommet';
 import { usePushToLogin } from '#helpers/routes';
 import UserCredentialsForm from '#shared/UserCredentialsForm';
 import { sendRecoveryEmail } from '#api/user-client';
 
 const RecoveryEmail = () => {
-  const sendMail = ({ email }) => sendRecoveryEmail(email);
+  const [visible, setVisible] = useState(false);
   const pushToLogin = usePushToLogin();
 
+  const sendMail = ({ email }) => {
+    return sendRecoveryEmail(email).then(() => {
+      setVisible(true);
+      // Return a promise that resolves after 3 seconds to delay the redirect
```

Comment:
this can be an extracted function in an utils module, it seems that it's code that can be reused